tls/s2n_client_cert.c

```diff
@@ -130,8 +131,15 @@ int s2n_client_cert_recv(struct s2n_connection *conn)
 
     /* Determine the Cert Type, Verify the Cert, and extract the Public Key */
     s2n_pkey_type pkey_type = S2N_PKEY_TYPE_UNKNOWN;
-    POSIX_GUARD_RESULT(s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn, cert_chain_data,
-            cert_chain_size, &pkey_type, &public_key));
+    s2n_result result = s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn, cert_chain_data,
+            cert_chain_size, &pkey_type, &public_key);
+    if (s2n_result_is_error(result)) {
+        /* reset the stuffer cursor to where cert data start in order to prepare for re-entry, because
+         * s2n_handle_retry_state() will invoke s2n_client_cert_recv() directly without updating stuffer
+         */
+        conn->handshake.io.read_cursor = cert_start;
```

Comment:
It's better to use the stuffer methods rather than directly modifying the internal stuffer state, as the the stuffer methods perform checks to make sure we don't accidentally set the cursors to an invalid position.
```suggestion
        POSIX_GUARD(s2n_stuffer_reread(&conn->handshake.io);
        POSIX_GUARD(s2n_stuffer_skip_read(&conn->handshake.io, cert_start));
```